typeid/typeid.py

```diff
@@ -80,15 +80,14 @@ def from_uuid(suffix: uuid6.UUID, prefix: Optional[str] = None) -> TypeID:
 
 
 def get_prefix_and_suffix(string: str) -> tuple:
-    parts = string.split("_")
-    prefix = None
+    parts = string.rsplit("_", 1)
     if len(parts) == 1:
+        prefix = None
         suffix = parts[0]
-    elif len(parts) == 2 and parts[0] != "":
-        suffix = parts[1]
-        prefix = parts[0]
-    else:
+    elif len(parts) == 2 and parts[0] == "":
         raise InvalidTypeIDStringException(f"Invalid TypeID: {string}")
+    else:
+        prefix, suffix = parts
```

Comment:
This doesn't work as it should. As you can see from the exception raised in the test,  it is not `InvalidTypeIDStringException` but `SuffixValidationException`.

It fails when you provide a string with multiple underscores. In that case `part[0]` will never be equal to an empty string and the prefix will be incorrect, in other words, we won't be able to throw an error. 

Better to use `split` and add another condition for length being 3 or more. 

```suggestion
    elif len(parts) >= 3:
        raise InvalidTypeIDStringException(f"Invalid TypeID: {string}")
    else:
        prefix, suffix = parts
```